sklearn/metrics/_classification.py

```diff
@@ -2371,10 +2371,18 @@ def hinge_loss(y_true, pred_decision, *, labels=None, sample_weight=None):
     y_true = column_or_1d(y_true)
     y_true_unique = np.unique(labels if labels is not None else y_true)
     if y_true_unique.size > 2:
-        if (labels is None and pred_decision.ndim > 1 and
-                (np.size(y_true_unique) != pred_decision.shape[1])):
-            raise ValueError("Please include all labels in y_true "
-                             "or pass labels as third argument")
+        if pred_decision.ndim <= 1 or \
+                y_true_unique.size != pred_decision.shape[1]:
+            if labels is not None or pred_decision.ndim <= 1:
+                raise ValueError("The shape of pred_decision is not "
+                                 "consistent with the number of classes. "
+                                 "pred_decision shape must be "
+                                 "(n_samples, n_classes) with "
+                                 "multiclass target")
+            else:
+                raise ValueError("Please include all labels in y_true "
+                                 "or pass labels as third argument")
```

Comment:
A small nit. With another condition, on `labels is None`, the logic becomes harder to follow at a glance. At this point, I think creating `invalid_decision_shape ` would make this condition more explicit: 

```python
invalid_decision_shape = (pred_decision.ndim > 1 and 
                          y_true_unique.size != pred_decision.shape[1]))

if labels is None:
    if invalid_decision_shape:
        raise ValueError("Please include all the labels...")
elif invalid_decision_shape or pred_decision.ndim <= 1:
    raise ValueError("The shape of pred_decision...")
```

Reply:
Ok, indeed, and it's more elegant.

Reply:
But we still have the same issue. 
If the user do like #19638, it means ``inge_loss(y_true=[2,1,0,1,0,1,1], pred_decision=[0,1,2,1,0,2,1])``, 
the user will get the message "Please include all the labels..." whereas the ture error should be "The shape of pred_decision...". 

The logical difficulty comes from the fact that there is a "specific case" for the situation multiclass problem AND ``pred_decision`` 1d given. It means that the user did not understand the meaning of this parameter (like I didn't)